File: packages/python/src/output_guard/strategies/strip_fences.py
"""strip-fences: Remove markdown code fences wrapping JSON/YAML/TOML/etc.

Mirror of packages/ts/src/strategies/strip-fences.ts.
"""

from __future__ import annotations

import re

_FENCE = re.compile(r"^```[\w-]*\r?\n([\s\S]*?)\r?\n```\s*$", re.IGNORECASE)
_FENCE_INLINE = re.compile(r"^```[\w-]*\r?\n([\s\S]*?)```\s*$", re.IGNORECASE)


def strip_fences(text: str) -> str:
    trimmed = text.strip()
    match = _FENCE.match(trimmed)
    if match is not None:
        return match.group(1).strip()
    inline = _FENCE_INLINE.match(trimmed)
    if inline is not None:
        return inline.group(1).strip()
    return text

File: packages/ts/src/strategies/strip-fences.ts
/**
 * strip-fences: Remove markdown code fences wrapping JSON/YAML/TOML/etc.
 * Handles ```json, ```yaml, ```toml, ``` (unlabelled).
 * Runs before extract-json.
 */
export const stripFences = (text: string): string => {
  const trimmed = text.trim();
  const match = /^```[\w-]*\r?\n([\s\S]*?)\r?\n```\s*$/i.exec(trimmed);
  if (match?.[1] !== undefined) {
    return match[1].trim();
  }
  // Inline fence without trailing newline before closing ```
  const inlineMatch = /^```[\w-]*\r?\n([\s\S]*?)```\s*$/i.exec(trimmed);
  if (inlineMatch?.[1] !== undefined) {
    return inlineMatch[1].trim();
  }
  return text;
};
